python/ray/serve/_private/http_proxy.py

```diff
@@ -417,6 +458,8 @@ async def __call__(self, scope, receive, send):
                 }
             )
 
+        self._ongoing_requests_end()
```

Comment:
This needs to be in a `finally` block. Otherwise, if there is any unexpected exception in the body of `__call__`, we could end up with a permanent count of >= 0 and never be able to drain this node